assets/components/contributionPaymentCtas/contributionPaymentCtas.jsx

```diff
@@ -0,0 +1,147 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+
+import type { Node } from 'react';
+
+import CtaLink from 'components/ctaLink/ctaLink';
+import ErrorMessage from 'components/errorMessage/errorMessage';
+
+import {
+  getSpokenType,
+  getOneOffSpokenName,
+} from 'helpers/contributions';
+import { generateClassName } from 'helpers/utilities';
+import { routes } from 'helpers/routes';
+import { addQueryParamsToURL } from 'helpers/url';
+
+import type { Currency } from 'helpers/internationalisation/currency';
+import type { Contrib as ContributionType } from 'helpers/contributions';
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import type { ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
+
+
+// ----- Types ----- //
+
+type PropTypes = {
+  contributionType: ContributionType,
+  amount: number,
+  referrerAcquisitionData: ReferrerAcquisitionData,
+  country: IsoCountry,
+  currency: Currency,
+  canClick: boolean,
+  PayPalButton: Class<React$Component<*>>,
+  error: ?string,
+  resetError: void => void,
+};
+
+
+// ----- Heading ----- //
+
+// Prevents a click event if it's not allowed.
+function onCtaClick(canClick: boolean, resetError: void => void): Function {
```

Comment:
Instead of `void => void` why don't we use `Function`? I think it is more readable or we need to restrict the function to be void => void?

Reply:
Function is more generic, it can literally be any Function, `void => void` applies a more specific type.